File: app/src/main/java/de/garten/training/depotflow/data/mapper/ChecklistMapper.java
package de.garten.training.depotflow.data.mapper;

import de.garten.training.depotflow.data.api.dto.ChecklistItemDto;
import de.garten.training.depotflow.data.db.green.ChecklistItem;

public class ChecklistMapper {

    public ChecklistItem fromDto(ChecklistItemDto dto) {
        ChecklistItem item = new ChecklistItem();
        item.setRemoteId(dto.id);
        item.setLabel(dto.label);
        item.setChecked(dto.checked);
        item.setMandatory(dto.mandatory);
        item.setNote(dto.note);
        return item;
    }
}

File: app/src/main/java/de/garten/training/depotflow/data/mapper/ChecklistMapper.kt
package de.garten.training.depotflow.data.mapper

import de.garten.training.depotflow.data.api.dto.ChecklistItemDto
import de.garten.training.depotflow.data.db.green.ChecklistItem

class ChecklistMapper {
    fun fromDto(dto: ChecklistItemDto): ChecklistItem {
        return ChecklistItem().apply {
            remoteId = dto.id
            label = dto.label
            isChecked = dto.checked
            isMandatory = dto.mandatory
            note = dto.note
        }
    }
}
